src/main/java/com/yahoo/bullet/pubsub/Metadata.java

```diff
@@ -5,21 +5,56 @@
  */
 package com.yahoo.bullet.pubsub;
 
-import lombok.AllArgsConstructor;
 import lombok.Getter;
 import lombok.NoArgsConstructor;
 import lombok.Setter;
 
 import java.io.Serializable;
 
-@Getter @Setter @AllArgsConstructor @NoArgsConstructor
+@NoArgsConstructor
 public class Metadata implements Serializable {
     public enum Signal {
         ACKNOWLEDGE,
-        COMPLETE
+        COMPLETE,
+        FAIL
     }
+
+    private static final long serialVersionUID = 4234800234857923112L;
+
+    @Getter @Setter
     private Signal signal;
-    private Serializable content;
+    private Object content;
```

Comment:
Why did you change from Serializable to Object and do the type cast after?

Reply:
If the type is Serializable, GSON cannot create an instance of it. So I kept the interface the same (allowing only Serializable objects in or out) but store it as an object. The casting is only done on the getter.  GSON is used in PubSubMessage#fromJSON. See a better way?

Reply:
Then add a comment here?

Reply:
Will do.